Tasks/NodeTool/nodetool.ts

```diff
@@ -0,0 +1,154 @@
+//import toolLib = require('vsts-task-tool-lib/tool');
+//import taskLib = require('vsts-task-lib/task');
+import * as toolLib from 'vsts-task-tool-lib/tool';
+import * as taskLib from 'vsts-task-lib/task';
+import * as restm from 'typed-rest-client/RestClient';
+import * as os from 'os';
+import * as path from 'path';
+
+let osPlat: string = os.platform();
+let osArch: string = os.arch();
+
+async function run() {
+    try {
+        let versionSpec = taskLib.getInput('versionSpec', true);
+        let onlyLTS: boolean = taskLib.getBoolInput('onlyLTS', false);
+
+        await getNode(versionSpec, onlyLTS);
+    }
+    catch (error) {
+        console.error('ERR:' + error.message);
+    }
+}
+
+//
+// Node versions interface
+// see https://nodejs.org/dist/index.json
+//
+interface INodeVersion {
+    version: string,
+    lts: any,
+    files: string[]
+}
+
+//
+// Basic pattern:
+//     version = find and evaluate local versions
+//          use latest match 
+//     if version not found locally
+//          // let's query
+//          if versionSpec is explicit version
+//               versionToGet = versionSpec
+//          else
+//               versionToGet = query and evaluate internet tool provider
+//          
+//          download versionToGet
+//          Extract or move to cache download
+//
+//      find tool path by version
+//      prepend $PATH with toolpath
+//
+async function getNode(versionSpec: string, onlyLTS: boolean) {
+    //
+    // Let's try and resolve the versions spec locally first
+    //
+    let localVersions: string[] = toolLib.findLocalToolVersions('node');
+    let version: string = toolLib.evaluateVersions(localVersions, versionSpec);
+
+    if (version) {
+        console.log('Resolved from tool cache: %s', version);
+    }
+    else {
+        //
+        // Let's query and resolve the latest version for the versionSpec
+        // If the version is an explicit version (1.1.1 or v1.1.1) then no need to query    
+        //
+        if (toolLib.isExplicitVersion(versionSpec)) {
+            // given exact version to get
+            toolLib.debug('explicit match', versionSpec);
+            version = versionSpec;
+        }
+        else {
+            // let's query for version
+            let versions: string[] = [];
+
+            // node offers a json list of versions
+            let dataFileName: string;
+            switch (osPlat) {
+                case "linux": dataFileName = "linux-" + osArch; break;
+                case "darwin": dataFileName = "osx-" + osArch + '-tar'; break;
+                case "win32": dataFileName = "win-" + osArch; break;
+            }
+
+            let dataUrl = "https://nodejs.org/dist/index.json";
+            let ltsMap : {[version: string]: string} = {};
```

Comment:
Technically this is not accurate. The value is sometimes a boolean and sometimes a string. Although I don't think it currently matters for the logic below.